### ISSUE TITLE
a question about loss in TwoLayerNet 

### ISSUE THREAD

**Issue by wtliao** [Burton2000/CS231n-2017#8]
Hi, thanks for sharing your solution.
But I have a question about the loss definition in neural_net.py line 117,118 as:

```python
correct_class_scores = np.choose(y, shift_scores.T) # Size N vector
loss = -correct_class_scores + np.log(np.sum(np.exp(shift_scores), axis=1))
```
why correct_class_scores is added to the loss? Thanks

**Comment by Burton2000** [comment]
Thanks for the question.

Basically from https://cs231n.github.io/linear-classify/#softmax 

There is two formulas for cross entropy loss given:
![ce](https://user-images.githubusercontent.com/17595269/58957892-902b1780-8799-11e9-85ff-8e9223574fb8.PNG)

The one on the left is involves calculating the softmax of the output scores, then the loss is the negative log of the correct classes softmax score.

I implemented the formula on the right which is why correct_class_scores is subtracted from the loss.

Looking at my code again it might be a bit unclear as I calculate the softmax scores for the network outputs but then dont actually use this anywhere. I might go back and change this to make it more clear. 

Hope that helps you.
